optuna/samplers/_grid.py

```diff
@@ -122,8 +125,9 @@ def __init__(
         self._all_grids = list(itertools.product(*self._search_space.values()))
         self._param_names = sorted(search_space.keys())
         self._n_min_trials = len(self._all_grids)
-        self._rng = LazyRandomState(seed)
-        self._rng.rng.shuffle(self._all_grids)  # type: ignore[arg-type]
+        self._rng = LazyRandomState(seed or 0)
+        if seed is not None:
+            self._rng.rng.shuffle(self._all_grids)  # type: ignore[arg-type]
```

Comment:
How about always shuffling? I think it is not proper to use the grid in order only when `seed=None`.

```suggestion
        if seed is None:
            seed = 0
        self._rng = LazyRandomState(seed)
        self._rng.rng.shuffle(self._all_grids)  # type: ignore[arg-type]
```